### PR TITLE
feat: Enhance address hierarchy configuration to support free text levels

### PR DESCRIPTION
**Pull request by reagan-meant**
- Updated CreatePatientForm to handle free text address hierarchy levels.
- Modified AddressHierarchyRestController to include allowFreeText property in address hierarchy levels.
- Adjusted AddressHierarchyConfigurationHandler to process allowFreeText from CSV and validate its usage.
- Implemented identity type registration for both coded and free text levels.
- Added unit tests for AddressHierarchyConfigurationHandler to cover new functionality and edge cases.
- Updated Indonesia address hierarchy CSV to reflect new structure with allowFreeText.

# Pull Requests Requirements

- [ ] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [ ] The PR title follows conventional commit label standards.
- [ ] The changes confirm to the OpenElis Global x3
      [Styleguide and Design](https://uwdigi.atlassian.net/wiki/spaces/OG/pages/621346838/OpenELIS+Global+Style+Guide)
      documentation.
- [ ] The changes include tests or are validated by existing tests.
- [ ] I have read and agree to the Contributing
      [Guidelines](https://openelis-global.org/community/get-involved/) of this
      project.

## Screenshots

[Add relevant screenshots here if applicable]

## Related Issue

[Add a link to the related issue or mention it here if applicable]

## Other

[Add any additional information or notes here]

### CODE REVIEW

**Comment by pmanko** [comment]
Pushed a merge of `develop` plus one follow-up fix as maintainer. CI is re-running.

**What landed on `OGC-317f`:**

- `3d3cd9484` — `Merge branch 'develop' into OGC-317f`. Clean textual merge. Git's rename detection correctly forwarded the `.js` address-hierarchy edits onto develop's Vite-renamed `CreatePatientForm.jsx`. The free-text conditional rendering and the cascade-skip logic are both preserved at the right call sites.
- `b5c889d84` — `fix(address-hierarchy): inject DisplayListService to unbreak tests after develop merge`. Develop's #3329 added `DisplayListService.getInstance().refreshLists()` to both `AddressHierarchy{,Values}ConfigurationHandler`. Your unit tests run under `MockitoJUnitRunner` with no Spring context, so the static singleton was null and 13 of 19 tests NPE'd after the merge — a classic semantic merge conflict that git can't detect textually. Fix: switched both handlers to standard Spring DI (matches the existing four `@Autowired` collaborators), added `@Mock DisplayListService` in the test, and added one `verify(displayListService).refreshLists()` assertion in the happy-path test to close a pre-existing coverage gap. No static-mocking, no dependency churn. 19/19 green locally.
- Submodules pinned to latest default-branch HEAD.

**CI at the moment:** backend Build + Test passing (13m20s), Shared Build passing (13m3s), Image/Static/Run/Frontend Checkpoint all green. E2E still running.

**Two asks before merge:**

1. **Release-note line for the CSV schema change.** `indonesia-levels.csv` drops `displayKey` and `sortOrder` columns. Deployments with custom `*-levels.csv` files will silently ignore those columns post-upgrade. Not a blocker — just worth surfacing in release notes.
2. **Nit (optional):** a one-line comment on the `ADDRESS_HIERARCHY_\\d+(_TEXT)?` regex in `PatientServiceImpl.persistIdentityTypes()` and `PatientManagementUpdate.persistIdentityTypes()`. The `_TEXT` suffix is a cross-stack contract between this regex and the frontend's `addressHierarchy_${levelIndex}_text` field name — worth being explicit about.

Will formalize approval via `gh pr review --approve` once E2E lands.

File: src/main/java/org/openelisglobal/address/service/AddressHierarchyConfigurationHandler.java [hunk 1]
```diff
@@ -90,9 +105,8 @@ public void processConfiguration(InputStream inputStream, String fileName) throw
         // Get column indices
         int levelIndex = findColumnIndex(headers, "level");
         int typeNameIndex = findColumnIndex(headers, "typeName");
-        int displayKeyIndex = findColumnIndex(headers, "displayKey");
-        int sortOrderIndex = findColumnIndex(headers, "sortOrder");
         int defaultValueIndex = findColumnIndex(headers, "defaultValue");
+        int allowFreeTextIndex = findColumnIndex(headers, "allowFreeText");
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/address/service/AddressHierarchyConfigurationHandler.java:109]: Heads-up: develop's handler now parses `inputType` and `bindKey` columns (added in #3553) and still supports `displayKey`/`sortOrder`, which this change removes. Rebasing will conflict throughout this method — the `allowFreeText` column would need to be reconciled with (or replaced by) the shipped `inputType=freetext` semantics.

File: src/main/java/org/openelisglobal/address/controller/AddressHierarchyRestController.java [hunk 2]
```diff
@@ -55,8 +55,9 @@ public List<AddressHierarchyLevel> getLevels() {
             if (level > 0) {
                 String defaultValue = getDefaultValueForLevel(level);
                 String defaultId = resolveDefaultValueToId(defaultValue, orgType.getName());
-                levels.add(
-                        new AddressHierarchyLevel(level, orgType.getId(), orgType.getName(), defaultValue, defaultId));
+                boolean allowFreeText = Boolean.TRUE.equals(orgType.getAllowFreeText());
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/address/controller/AddressHierarchyRestController.java:58]: On current develop, `AddressHierarchyLevel` already carries `inputType`, `displayKey`, `sortOrder`, and `bindKey`, and the frontend switches on `level.inputType === "freetext"`. Adding a parallel `allowFreeText` boolean would give the API two overlapping ways to express the same thing — worth consolidating on one.

File: src/main/resources/liquibase/3.4.x.x/019-add-organization-type-allow-free-text.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+
+    <changeSet author="reagan-meant" id="add-allow-free-text-column">
```
Comment (mozzy11) [inline src/main/resources/liquibase/3.4.x.x/019-add-organization-type-allow-free-text.xml:7]: If this schema change is still needed after reconciling with #3553, it should live in the current `3.5.x.x` changelog stream (latest changeset there is `039-test-method-links.xml`) rather than `3.4.x.x`, which corresponds to an already-released version.

**Review by mozzy11 — COMMENTED** [review]
Thanks @reagan-meant — this is a well-structured end-to-end implementation (Liquibase precondition, coded-values guard, eager identity-type registration, and a thorough unit-test suite), and CI is green on the branch.

However, develop has since shipped free-text address hierarchy levels via #3553: the levels CSV now supports `inputType` (`dropdown`/`freetext`) plus `bindKey`, persisted in `site_information` (`AddrHierarchyInputType_N` / `AddrHierarchyBindKey_N`), exposed through `AddressHierarchyRestController`, and rendered as a `TextInput` in `CreatePatientForm.tsx` (`level.inputType === "freetext"`). Handler unit tests also exist now (`AddressHierarchyConfigurationHandlerMetadataTest`, `AddressHierarchyConfigurationHandlerNormalizeInputTypeTest`). That covers the core goal of this PR through a different mechanism, and several of this PR's design choices now conflict directly: the CSV format here drops `displayKey`/`sortOrder`, which develop kept and extended; and `CreatePatientForm.jsx` was migrated to TypeScript in #3872, so all frontend hunks target a file that no longer exists.

A rebase would be major: every backend file touched here has moved (#3553, #3567, #3591, #3356), the frontend must be re-implemented in `.tsx`, and the branch also carries unrelated `dataexport`/`plugins` submodule pointer bumps that should be dropped.

Could you confirm what (if anything) from this PR is still needed beyond what #3553 delivered — e.g. persisting the flag on `organization_type` rather than `site_information`, the `ADDRESS_HIERARCHY_N_TEXT` identity-type persistence path, or the `OrganizationType` HBM-to-annotations migration? If there is a remaining gap, a fresh, focused PR on top of current develop would likely be easier than rebasing this one.

File: frontend/src/components/patient/CreatePatientForm.jsx [hunk 4]
```diff
@@ -1346,55 +1357,80 @@ function CreatePatientForm(props) {
...
-                                        e.target.value,
-                                      );
-                                      handleAddressHierarchySelection(
-                                        levelIndex,
-                                        e.target.value,
-                                        setFieldValue,
-                                      );
-                                      // For backward compatibility, also set healthRegion/healthDistrict
-                                      if (levelIndex === 0) {
+                              {level.allowFreeText ? (
```
Comment (mozzy11) [inline frontend/src/components/patient/CreatePatientForm.jsx:1360]: `CreatePatientForm.jsx` was migrated to TypeScript (`CreatePatientForm.tsx`) in #3872, and the `.tsx` version already renders free-text levels via `level.inputType === "freetext"` with a `bindKey`-bound `TextInput`. These hunks would need to be re-implemented against the TSX file if anything here is still required.
